zap/src/main/java/org/zaproxy/zap/extension/ascan/FilterPanel.java

```diff
@@ -0,0 +1,503 @@
+            excTagList = new JList<>(getTagModel());
+            excTagList.setPrototypeCellValue("Tags are short...");
+
+            excTagScroller = new JScrollPane(excTagList);
+            excTagScroller.setHorizontalScrollBarPolicy(
+                    javax.swing.JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+            excTagScroller.setVerticalScrollBarPolicy(
+                    javax.swing.JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED);
+        }
+        return excTagScroller;
+    }
+
+    public void setAllTags(List<String> allTags) {
+        List<String> selected = incTagList.getSelectedValuesList();
+        int[] inds = new int[allTags.size()];
+        Arrays.fill(inds, -1);
+
+        getTagModel().clear();
+        int i = 0;
+        for (String tag : allTags) {
+            getTagModel().addElement(tag);
+        }
+        for (Object sel : selected) {
+            if (getTagModel().contains(sel)) {
+                inds[i] = getTagModel().indexOf(sel);
+            }
+            i++;
+        }
+        incTagList.setSelectedIndices(inds);
+    }
+
+    private void addTagsInFilterPanel(SiteNode siteNode, Collection<String> tags) {
+        if (siteNode != null) {
+            if (siteNode.getHistoryReference() != null
+                    && siteNode.getHistoryReference().getTags() != null) {
+                tags.addAll(siteNode.getHistoryReference().getTags());
+            }
+            Enumeration<?> childEnumeration = siteNode.children();
+            while (childEnumeration.hasMoreElements()) {
+                Object node = childEnumeration.nextElement();
+                if (node instanceof SiteNode) {
+                    addTagsInFilterPanel((SiteNode) node, tags);
+                }
+            }
+        }
+    }
+
+    private void populateTagsInFilterPanel(Target target) {
+        Set<String> tags = new LinkedHashSet<>();
+        if (target != null) {
+            SiteNode siteNode = target.getStartNode();
+            if (siteNode != null) {
+                this.addTagsInFilterPanel(siteNode, tags);
+            } else if (target.getContext() != null || target.isInScopeOnly()) {
+                List<SiteNode> nodes = Collections.emptyList();
+                if (target.isInScopeOnly()) {
+                    nodes = Model.getSingleton().getSession().getTopNodesInScopeFromSiteTree();
+                } else if (target.getContext() != null) {
+                    nodes = target.getContext().getTopNodesInContextFromSiteTree();
+                }
+                for (SiteNode node : nodes) {
+                    this.addTagsInFilterPanel(node, tags);
+                }
+            }
+        }
+        this.setAllTags(new ArrayList<String>(tags));
+    }
+
+    public String validateFields() {
+        try {
+            strToRegexList(regexInc.getText());
+            strToRegexList(regexExc.getText());
+        } catch (PatternSyntaxException e1) {
+            // Invalid regex
```

Comment:
Comment can be removed (the catch block already says it).

Reply:
sure.